### ISSUE TITLE
Switching kubectl context while debugging causes cleanup to run against the incorrect context

### ISSUE THREAD

**Issue by peter-crist** [GoogleCloudPlatform/cloud-code-vscode#673]
**Version information**

Cloud Code Extension version: v1.20.3

VSCode version: 1.69.2 (system setup)

OS: Windows 10 Pro/WSL Ubuntu 20.04

Skaffold: v1.39.2

Kubectl: 
```
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.9", GitCommit:"c1de2d70269039fe55efb98e737d9a29f9155246", GitTreeState:"clean", BuildDate:"2022-07-13T14:26:51Z", GoVersion:"go1.17.11", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22+", GitVersion:"v1.22.13-eks-15b7512", GitCommit:"94138dfbea757d7aaf3b205419578ef186dd5efb", GitTreeState:"clean", BuildDate:"2022-08-31T19:15:48Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}
```

**Description**:
Switching kubectl contexts while actively debugging will cause Skaffold to cleanup resources associated with the wrong context when you stop the debugging session.

**Repro step**:
I was running Cloud Code to deploy to my local kind cluster starting a debugging session. While debugging, my kubectl context was changed and pointed to a development cluster (not local). I then stopped my Cloud Code debugging session. The cleanup that occurred as part of stopping the debugging session causes skaffold to delete resources in the incorrect context.

I would expect the cleanup to be executed against the context with which the debugging session was started. This caused all resources in a shared development cluster to be deleted.

While we can (and will) protect this with skaffold profiles - Cloud Code should not automatically cleanup if it notices the context was changed.

**Comment by SKrupa** [comment]
Thank you for reporting this bug, we'll investigate what we can do in this instance. We should be able to hold onto the old context to do the proper cleanup, but at the very least we shouldn't be deleting resources on a cluster other than the original.

**Comment by SKrupa** [comment]
A fix for this was released as part of the 1.21.6 release, actions should now all be executed against the original active cluster when the deployment was launched, not the current active cluster. Please let us know if you still face any unexpected behavior.
